CoreFoundation/NumberDate.subproj/CFTimeZone.c

```diff
@@ -183,11 +185,133 @@ static void __CFTimeZoneGetOffset(CFStringRef timezone, int32_t *offset) {
 
     RegCloseKey(hKey);
 }
+#elif TARGET_OS_ANDROID
+typedef Boolean (*__CFAndroidTimeZoneListEnumerateCallback)(const char name[40], int32_t start, int32_t length, FILE *fp, void *context);
+
+static void __CFAndroidTimeZoneListEnumerate(__CFAndroidTimeZoneListEnumerateCallback callback, void *context) {
+    // The best reference should be Android Bionic's libc/tzcode/bionic.cpp
+    static const char *tzDataFiles[] = {
+        "/data/misc/zoneinfo/current/tzdata",
+        "/system/usr/share/zoneinfo/tzdata"
+    };
+
+    Boolean done = FALSE;
+    for (int idx = 0; idx < sizeof(tzDataFiles)/sizeof(tzDataFiles[0]) && !done; idx++) {
+        FILE *fp = fopen(tzDataFiles[idx], "rb");
+        if (!fp) {
+            continue;
+        }
+
+        char header[24];
```

Comment:
Can you add a blob explaining the structure to the beginning?

Reply:
I will try to write my understanding of the file format.